test/dnn/miopen.jl

```diff
@@ -0,0 +1,90 @@
+using AMDGPU.MIOpen
+
+const MIOPEN_TYPES = (
+    (Float16, MIOpen.miopenHalf),
+    (Float32, MIOpen.miopenFloat),
+    (Float64, MIOpen.miopenDouble),
+    (Int8, MIOpen.miopenInt8),
+    (Int32, MIOpen.miopenInt32),
+)
+const MIOPEN_CONV_TYPES = (Float16, Float32) # FIXME: support more types
```

Comment:
Maybe we can move these type definitions into AMDGPU.jl, instead of just defining them in the tests?